### PR TITLE
Award year implementation.

### PR DESCRIPTION
**Pull request by jaleszek**

### CODE REVIEW

File: app/models/award_year.rb [hunk 1]
```diff
@@ -9,8 +9,24 @@ class AwardYear < ActiveRecord::Base
...
+  def self.mock_current_year?
+    Rails.env.test?
+  end
+
   def self.current
-    for_year(Date.current.year + 1).first_or_create
+    return where(year: 2016).first_or_create if mock_current_year?
+
+    now = DateTime.now
+    deadline = AwardYear.where(year: now.year + 1).first_or_create.settings.deadlines.submission_start.try(:trigger_at)
```
Comment (matthewford) [inline app/models/award_year.rb]: Line is too long. [119/100]

File: spec/models/award_year_spec.rb [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      context "22 of April 2016" do
+        it "is awarding year 2017 (takes 21.04 as default)" do
+          Timecop.freeze(Date.new(2016, 4, 22)) do
+            expect(described_class.current.year).to eq(2017)
+          end
+        end
+      end
+    end
+
+    context "submission begining period is 18 of July 2015, next submission begining 14 of Feb 2016" do
```
Comment (matthewford) [inline spec/models/award_year_spec.rb]: Line is too long. [103/100]

File: spec/models/form_answer_spec.rb [hunk 3]
```diff
@@ -23,12 +23,15 @@
 
   context "URN" do
     before do
-      FormAnswer.connection.execute("ALTER SEQUENCE urn_seq_#{Date.today.year + 1} RESTART")
-      FormAnswer.connection.execute("ALTER SEQUENCE urn_seq_promotion_#{Date.today.year + 1} RESTART")
+      FormAnswer.connection.execute("ALTER SEQUENCE urn_seq_#{AwardYear.current.year} RESTART")
+      FormAnswer.connection.execute("ALTER SEQUENCE urn_seq_promotion_#{AwardYear.current.year} RESTART")
```
Comment (matthewford) [inline spec/models/form_answer_spec.rb]: Line is too long. [105/100]

**Comment by jaleszek** [comment]
I need to test one thing more, let's keep it hardcoded for now

**Pull request closed without merging**